### PR TITLE
[FIX] html_editor: avoid crash on tables without a tbody

### PR DESCRIPTION
**Pull request by micheledic**
### Before this PR
Opening a record with a content that contains a table with no `<tbody>` (for example a table that was pasted/imported and left with only a caption or header) crashes the editor: the field never renders and the page throws an OwlError instead.

Steps to reproduce:
1. Open a record with an html field (e.g. a product description, or a knowledge article).
2. Insert/paste a table that has no body rows.
3. Reload the record.
4. The editor fails to mount — the field is blank and the console shows an OwlError.

### After this PR
If there is not tbody, the error doesn't appear

Bug introduced by https://github.com/odoo/odoo/pull/272977 
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by micheledic** [comment]
@sbbh-odoo

**Comment by sbbh-odoo** [comment]
Hello @micheledic, how did you manage to copy content with a table without a `<tbody>`? I didn't know we could end up with such a structure.
Thanks for the fix, it looks good to me. I also don't think there's a need to add the first test, as this case is already covered by many existing tests.

**Comment by micheledic** [comment]
> Hello @micheledic, how did you manage to copy content with a table without a `<tbody>`? I didn't know we could end up with such a structure. Thanks for the fix, it looks good to me. I also don't think there's a need to add the first test, as this case is already covered by many existing tests.

In my case, I received an email without a `<tbody>` element, which created a ticket. When I opened the ticket, the error appeared.

Are you sure that a test is not required to prevent regressions? Before my PR, all tests were green despite this edge case not being handled.

**Comment by sbbh-odoo** [comment]
@micheledic I was referring to the first test case only. I think it's already covered by many existing tests, so I don't think we need an additional test for that specific part.

**Comment by micheledic** [comment]
> @micheledic I was referring to the first test case only. I think it's already covered by many existing tests, so I don't think we need an additional test for that specific part.

done

File: addons/html_editor/static/src/main/table/table_plugin.js [hunk 1]
```diff
@@ -187,6 +187,9 @@ export class TablePlugin extends Plugin {
         // Move table width and margin to tbody to prevent scrollbars on the editable.
         this.editable.querySelectorAll("table").forEach((table) => {
             const tBody = table.tBodies[0];
+            if (!tBody) {
+                return;
+            }
```
Comment (Zynton) [inline addons/html_editor/static/src/main/table/table_plugin.js:192]: I'm not sure we should allow empty tables to begin with because it would be impossible to handle (as we see here, we can't properly size/style it, and how would you then add a row or column to it?).
This is a functional question requiring @lebl-odoo 's confirmation, but I would instead add a row and a column by default when pasting an empty table (check clipboard plugin).

File: addons/html_editor/static/tests/table/misc.test.js [hunk 2]
```diff
@@ -313,3 +313,14 @@ describe("selected cell color in toolbar", () => {
...
 });
+
+describe("table width on setup", () => {
+    test("does not crash on a table without a tbody", async () => {
+        // A rowless table has no tbody element (tBodies[0] is undefined). The
+        // width-to-tbody move on setup must skip it instead of throwing.
+        const { editor } = await setupEditor(
+            `<table class="table table-bordered o_table" style="width: 500px;"><caption>c</caption></table>`
+        );
+        expect(editor.isDestroyed).toBe(false);
```
Comment (Zynton) [inline addons/html_editor/static/tests/table/misc.test.js:324]: Either way, I would also check the content of the editable. With my suggested change, it would become something like:
```html
<table class="table table-bordered o_table"><caption>c</caption><tbody style="width: 500px;"><tr><td>[]<br></td></tr></tbody></table>
```

File: addons/html_editor/static/src/main/table/table_plugin.js (hunk 1 above)
Comment (micheledic) [reply addons/html_editor/static/src/main/table/table_plugin.js:192]: The issue comes also when receiving mail and the content of the mail wthout tbody goes to an html field. We can't override everything to add tbody

File: addons/html_editor/static/tests/table/misc.test.js (hunk 2 above)
Comment (micheledic) [reply addons/html_editor/static/tests/table/misc.test.js:324]: As i said, the issue is not caused only while copy-pasting tables but also receiving broken table by email

File: addons/html_editor/static/src/main/table/table_plugin.js (hunk 1 above)
Comment (Zynton) [reply addons/html_editor/static/src/main/table/table_plugin.js:192]: The code you modify is already correcting the DOM when setting up. We can actually override everything to add tbody, and I believe we should. A table in the editor should always have tbody, at least one tr and at least one td.

**Comment by micheledic** [comment]
@sbbh-odoo what do you think ? which approach should we follow?

**Comment by sbbh-odoo** [comment]
As @Zynton mentioned, every table in the editor should have a `<tbody>`, since all table operations rely on it.
I also got a few tickets regarding tables containing only a `<thead>`, so I prepared a [PR](https://github.com/odoo/odoo/pull/276588) that normalizes the table structure during setup and on paste.

**Comment by Zynton** [comment]
Closed in favor of https://github.com/odoo/odoo/pull/276588. Thanks for your work!
